Controller test

hideotty opened the pull request:
# What
コントローラーテスト

# Why
必須機能の為

hideotty committed:
controllerのbefore_actionの記述修正

一部マージ前に戻ってしまった為、再度変更点を記述。message controllerのアクションの記述をまとめた。

hideotty committed:
rails-controller-testing導入

fakerの記述場所も修正した

hideotty committed:
factory_girlのgroup記述を作成

hideotty committed:
factory girlのuser 記述

password_confirmation部分が正しくなるように変数でパスワードを設定

hideotty committed:
require rails_helper記述追加

hideotty committed:
controller testの設定

hideotty committed:
ControllerMacros導入

login_userの設定を行った

hideotty committed:
Add controller test

ログインしている場合
アクション内で定義しているインスタンス変数があるか
該当するビューが描画されているか
ログインしていない場合
意図したビューにリダイレクトできているか
メッセージを作成するアクション

ログインしているかつ、保存に成功した場合
メッセージの保存はできたのか
意図した画面に遷移しているか
ログインしているが、保存に失敗した場合
メッセージの保存は行われなかったか
意図したビューが描画されているか

TOMARUMARU reviewed (commented):
テスト結果のキャプチャを貼ってください！

・ログインしているかつ、保存に成功した場合、意図した画面に遷移しているのか
・ログインしているが、保存に失敗した場合、メッセージの保存は行われなかったか

のテストがないように思うので、そのテストの実装もよろしくお願いします。

hideotty commented:
https://gyazo.com/137e488c24c6701159bbc7a32a534f27

hideotty committed:
controller testの記述修正

下記の点を修正
・ログインしているかつ、保存に成功した場合、意図した画面に遷移しているのか
・ログインしているが、保存に失敗した場合、メッセージの保存は行われなかったか

hideotty committed:
空行の追加、test内容の記述修正

空行→見やすくする為
「ログインしているかつ、保存に成功した場合、意図した画面に遷移しているのか」の記述がなかった為、記述を変更追加
